tests/model/utils/insertcontent.js

```diff
@@ -407,6 +407,25 @@ describe( 'DataController utils', () => {
 						'<paragraph>b</paragraph>'
 					);
 				} );
+
+				it( 'should not merge a paragraph wrapped in blockQuote with lists', () => {
```

Comment:
Just a single test? Please be more creative. E.g. there should be separate tests for left merge and right merge. This gives at least 3 tests. And I'm sure you can think of more.